### PR TITLE
fix(web): prevent duplicate chat input echo

### LINKED ISSUE #859: When typing into the interactive chat for fbeast the characters multi print to the screen
and when i hit the backspace all of the extras disappear.

### PR DESCRIPTION
**Pull request by djm204**
## Summary
- Preserve the chat Composer instance with a stable session-scoped key.
- Only add optimistic local user messages when the live WebSocket is open; HTTP fallback now waits for the server transcript before rendering the sent text.
- Add a regression test covering the fallback path so duplicate echo behavior does not return.

## Test Plan
- npm run build --workspace @franken/types
- npm --workspace @franken/web run test -- src/hooks/use-chat-session.test.tsx -t "does not optimistically echo fallback HTTP messages"
- npm --workspace @franken/web run test
- npm --workspace @franken/web run typecheck
- npm --workspace @franken/web run build
- npm run lint:any

Closes #859

### CODE REVIEW

**Comment by djm204** [comment]
@codex review

**Comment by djm204** [comment]
@codex review

**Comment by djm204** [comment]
@codex review

**Comment by djm204** [comment]
@codex review

**Comment by djm204** [comment]
@codex review

File: packages/franken-web/src/components/chat-shell.tsx [hunk 1]
```diff
@@ -912,6 +912,7 @@ export function ChatShell({ baseUrl, projectId, sessionId, version }: ChatShellP
                 showTypingIndicator={showTypingIndicator}
               />
               <Composer
+                key={activeSessionId ?? selectedSessionId ?? 'anonymous'}
```
[the remark this answers is not in the document]
Comment (djm204) [reply packages/franken-web/src/components/chat-shell.tsx]: Current head preserves the composer key across initial anonymous session boot so drafts typed while the session id is created are not lost.

File: packages/franken-web/src/hooks/use-chat-session.ts [hunk 2]
```diff
@@ -793,7 +797,10 @@ export function useChatSession(opts: UseChatSessionOptions): UseChatSessionResul
           setCostUsd(refreshed.costUsd);
           setStatus('idle');
         } catch (error) {
-          setMessages((current) => updateReceipt(current, clientMessageId, 'accepted'));
+          setMessages((current) => [
+            ...current.filter((message) => !(message.role === 'user' && message.receipt === 'failed' && message.content === content)),
+            { ...optimisticMessage, receipt: 'accepted' },
+          ]);
```
[the remark this answers is not in the document]
Comment (djm204) [reply packages/franken-web/src/hooks/use-chat-session.ts:823]: Current head only keeps the optimistic local row for live WebSocket sends and removes same-id local rows before accepted fallback insertion, avoiding fallback duplicates.

File: packages/franken-web/src/components/chat-shell.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (djm204) [reply packages/franken-web/src/components/chat-shell.tsx]: Fixed in 4abfcf2d/current head: anonymous composer key remains stable during new-session initialization, so session id boot no longer remounts the composer and drops drafts.

File: packages/franken-web/src/hooks/use-chat-session.ts [hunk 3]
```diff
@@ -763,21 +763,25 @@ export function useChatSession(opts: UseChatSessionOptions): UseChatSessionResul
...
-        role: 'user',
-        content,
-        timestamp: new Date().toISOString(),
-        receipt: 'sending',
-      },
-    ]);
+    if (optimisticAdd) {
+      setMessages((current) => [
+        ...current.filter((message) => !(message.role === 'user' && message.receipt === 'failed' && message.content === content)),
+        optimisticMessage,
```
[the remark this answers is not in the document]
Comment (djm204) [reply packages/franken-web/src/hooks/use-chat-session.ts:790]: Fixed in 4abfcf2d/current head: failed HTTP fallback sends now insert a retryable failed local message instead of marking a missing optimistic row.

File: packages/franken-web/src/hooks/use-chat-session.ts [hunk 4]
```diff
@@ -763,21 +763,25 @@ export function useChatSession(opts: UseChatSessionOptions): UseChatSessionResul
...
+      ]);
```
[the remark this answers is not in the document]
Comment (djm204) [reply packages/franken-web/src/hooks/use-chat-session.ts:791]: Fixed in 4abfcf2d/current head: HTTP fallback failures now add a failed user row with retry metadata so the resend control remains available.

File: packages/franken-web/src/components/chat-shell.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (djm204) [reply packages/franken-web/src/components/chat-shell.tsx]: Fixed on current head: unsaved/new conversations use the session seed based composer key, so activeSessionId assignment does not remount the draft.

File: packages/franken-web/src/hooks/use-chat-session.ts [hunk 5]
```diff
@@ -763,21 +763,23 @@ export function useChatSession(opts: UseChatSessionOptions): UseChatSessionResul
...
     setMessages((current) => [
       ...current.filter((message) => !(message.role === 'user' && message.receipt === 'failed' && message.content === content)),
-      {
-        id: clientMessageId,
-        role: 'user',
-        content,
-        timestamp: new Date().toISOString(),
-        receipt: 'sending',
-      },
+      optimisticMessage,
```
[the remark this answers is not in the document]
Comment (djm204) [reply packages/franken-web/src/hooks/use-chat-session.ts]: Fixed in 4abfcf2d/current head: fallback HTTP sends no longer append the optimistic row while pending; they wait for refresh success, accepted fallback, or failed fallback handling.

File: packages/franken-web/src/components/chat-shell.tsx [hunk 6]
```diff
@@ -355,12 +356,16 @@ export function ChatShell({ baseUrl, projectId, sessionId, version }: ChatShellP
...
+  const composerSessionKey = preserveComposerDraft
+    ? `anonymous:${sessionSeed}`
+    : selectedSessionId ?? activeSessionId ?? `anonymous:${sessionSeed}`;
+
   useEffect(() => {
-    if (!activeSessionId) {
+    if (!activeSessionId || preserveComposerDraft) {
       return;
     }
     setSelectedSessionId(activeSessionId);
```
[the remark this answers is not in the document]
Comment (djm204) [reply packages/franken-web/src/components/chat-shell.tsx]: Current head keeps the selectedSessionId guard before auto-syncing activeSessionId, so user-selected conversations are not overwritten by stale active session state.

File: packages/franken-web/src/components/chat-shell.tsx [hunk 7]
```diff
@@ -355,12 +356,16 @@ export function ChatShell({ baseUrl, projectId, sessionId, version }: ChatShellP
...
   }, [baseUrl]);
 
+  const composerSessionKey = preserveComposerDraft
+    ? `anonymous:${sessionSeed}`
+    : selectedSessionId ?? activeSessionId ?? `anonymous:${sessionSeed}`;
+
   useEffect(() => {
-    if (!activeSessionId) {
+    if (!activeSessionId || preserveComposerDraft || selectedSessionId) {
       return;
```
[the remark this answers is not in the document]
Comment (djm204) [reply packages/franken-web/src/components/chat-shell.tsx]: Fixed in 4abfcf2d/current head: default chat shells still sync the resolved activeSessionId into selectedSessionId for beast launch metadata while preserving composer drafts via the stable anonymous key.

File: packages/franken-web/src/hooks/use-chat-session.ts [hunk 8]
```diff
@@ -793,7 +795,10 @@ export function useChatSession(opts: UseChatSessionOptions): UseChatSessionResul
           setCostUsd(refreshed.costUsd);
           setStatus('idle');
         } catch (error) {
-          setMessages((current) => updateReceipt(current, clientMessageId, 'accepted'));
+          setMessages((current) => [
+            ...current.filter((message) => !(message.role === 'user' && message.receipt === 'failed' && message.content === content)),
+            { ...optimisticMessage, receipt: 'accepted' },
+          ]);
```
[the remark this answers is not in the document]
Comment (djm204) [reply packages/franken-web/src/hooks/use-chat-session.ts:823]: Fixed in 4abfcf2d/current head: accepted fallback refresh failures replace any same-id local row before appending the accepted fallback message, preventing duplicate accepted rows.

File: packages/franken-web/src/hooks/use-chat-session.ts [hunk 9]
```diff
@@ -763,21 +763,23 @@ export function useChatSession(opts: UseChatSessionOptions): UseChatSessionResul
...
       ...current.filter((message) => !(message.role === 'user' && message.receipt === 'failed' && message.content === content)),
-      {
-        id: clientMessageId,
-        role: 'user',
-        content,
-        timestamp: new Date().toISOString(),
-        receipt: 'sending',
-      },
+      optimisticMessage,
     ]);
```
[the remark this answers is not in the document]
Comment (djm204) [reply packages/franken-web/src/hooks/use-chat-session.ts]: Fixed in 4abfcf2d/current head: optimistic local messages are gated on a live socket; HTTP fallback pending sends no longer echo in the transcript while the composer still contains the draft.

**Comment by djm204** [comment]
@codex review

File: packages/franken-web/src/hooks/use-chat-session.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (djm204) [reply packages/franken-web/src/hooks/use-chat-session.ts:791]: Fixed in 37c766f1 by removing same-content failed user drafts before both HTTP fallback success paths (snapshot merge and refresh-failed accepted fallback), and added a regression test for composer resubmission after an HTTP retry succeeds while refresh fails.

**Comment by djm204** [comment]
@codex review

File: packages/franken-web/src/hooks/use-chat-session.ts [hunk 10]
```diff
@@ -763,37 +767,44 @@ export function useChatSession(opts: UseChatSessionOptions): UseChatSessionResul
...
           setPendingApproval(refreshed.pendingApproval ?? null);
           setTokenTotals(refreshed.tokenTotals);
           setCostUsd(refreshed.costUsd);
           setStatus('idle');
         } catch (error) {
-          setMessages((current) => updateReceipt(current, clientMessageId, 'accepted'));
+          setMessages((current) => [
+            ...current.filter((message) => message.id !== clientMessageId && !isFailedUserDraftForContent(message, content)),
+            { ...optimisticMessage, receipt: 'accepted' },
+          ]);
```
[the remark this answers is not in the document]
Comment (djm204) [reply packages/franken-web/src/hooks/use-chat-session.ts:823]: Fixed in e02a6311 by tracking the current active session id in a ref and no-oping stale HTTP fallback completions/failures before appending synthetic accepted or failed drafts.

File: packages/franken-web/src/components/chat-shell.tsx [hunk 11]
```diff
@@ -355,12 +356,16 @@ export function ChatShell({ baseUrl, projectId, sessionId, version }: ChatShellP
...
+    : selectedSessionId ?? activeSessionId ?? `anonymous:${sessionSeed}`;
+
   useEffect(() => {
-    if (!activeSessionId) {
+    if (!activeSessionId || selectedSessionId) {
       return;
     }
     setSelectedSessionId(activeSessionId);
-  }, [activeSessionId]);
+  }, [activeSessionId, selectedSessionId]);
```
[the remark this answers is not in the document]
Comment (djm204) [reply packages/franken-web/src/components/chat-shell.tsx]: Fixed in e02a6311 by making the auto-select effect respect preserveComposerDraft, so New conversation can keep selectedSessionId unset instead of immediately restoring the previous active session.

**Comment by djm204** [comment]
@codex review
